### PR TITLE
Example - Custom post types

### PR DESCRIPTION
**Pull request by kmaric69**
### Changelog

- Added example of how to register a custom post type.
- added menu item `Examples` to the `libs` sidebar

This PR addresses #8

### CODE REVIEW

File: docs/libs/examples/libs-example-custom-post-type.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+---
+id: libs-example-custom-post-type
+title: Register Custom Post Type
+---
+
+While you are working on your WordPress project it is possible that at one point you will need a Custom Post Type to enrich your project.
```
Comment (dingo-d) [inline docs/libs/examples/libs-example-custom-post-type.md]:
```suggestion
When working on a WordPress project it is possible that, at one point, you will need to register a Custom Post Type.
```

File: docs/libs/examples/libs-example-custom-post-type.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+For that case we have prepared an easy way for you to register a new post type following these steps and with the following code example:
```
Comment (dingo-d) [inline docs/libs/examples/libs-example-custom-post-type.md]:
```suggestion
In that case, we have prepared an easy way for you to register a new post type by following these steps with the following code example:
```

File: docs/libs/examples/libs-example-custom-post-type.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+```php
+
+* src
+  * class-main.php
+  * custom-post-type
+    * class-example.php
+
+```
+PRO HINT: Files containing a class should be named class-{classname}.php and there should always be only one class per file.
```
Comment (dingo-d) [inline docs/libs/examples/libs-example-custom-post-type.md]:
```suggestion
Files containing a class should be named `class-{classname}.php` and there should always be only one class per file.
```

I'd avoid a PRO word here. I think the PSR are using one class per file suggestion, and `class-{classname}.php` is a WordPress Coding Standard suggestion.

File: docs/libs/examples/libs-example-custom-post-type.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+## Follow these steps:
+* Create a new folder inside the `src` folder called `custom-post-type`. This is the place where you will hold all your classes for registering custom post types, Or put this in a place you think it would suit your project best.
+* Let's call our first custom post type `Projects`, create a file called the same as your custom post type, in this case `class-projects.php`.
```
Comment (dingo-d) [inline docs/libs/examples/libs-example-custom-post-type.md]:
```suggestion
* Let's call our first custom post type `Projects`. Create a file called the same as your custom post type, in this case, `class-projects.php`.
```

**Comment by iruzevic** [comment]
Awesome great work

File: docs/libs/examples/libs-example-custom-post-type.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+  /**
+   * Post type slug constant.
+   *
+   * @var string
+   */
+  const POST_TYPE_SLUG = 'projects-type';
+
+  /**
+   * Pretty slug for the custom post type.
```
Comment (gabriel-glo) [inline docs/libs/examples/libs-example-custom-post-type.md]:
```suggestion
   * URL slug for the custom post type.
```

**Pull request merged**